src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs

```diff
@@ -2,20 +2,18 @@
 using BudgetManager.Shared.DataAccess.MongoDB.DatabaseSettings;
 using BudgetManager.Shared.Models.MongoDB.Models.Interfaces;
 using MongoDB.Bson;
-using MongoDB.Bson.Serialization;
-using MongoDB.Bson.Serialization.Serializers;
 using MongoDB.Driver;
 using MongoDbGenericRepository.Attributes;
 
 namespace BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation
 {
-    public class BaseRepository<TDocument> : IBaseRepository<TDocument>
+    public abstract class BaseRepository<TDocument>
```

Comment:
you can inherit it and use its methods, it case if they not full fill business requirements, then you can override any

Reply:
According to the SOLID principles
In case of abstract class we will broke Dependency Inversion principle. High-level modules should not depend on lower-level modules.  Both should depend on abstractions. Abstractions should not depend on details. Details should depend on abstractions. 

In this case our repos shouldn't depend on the lower-level class (even if it's an abstract class) to get its work done. Both the lower-level and higher-level class should depend on the interface which defines how they are going to interact.

Our BaseRep should have only common things to implement, if we need smth to add - we should add it in our rep which will be inherited from BaseRep

We should have loosely coupled entities that are easy to test, modify and update.

Reply:
I see,
Ok, let me fix that